### DISCUSSION TITLE
Script to restart station on specific interval?

### DISCUSSION THREAD

**Discussion by unudelaradio** [AzuraCast/AzuraCast#7475] [Q&A]
Anyone could help me with some Liquidsoap Custom Config Script which auto restarts the station on specific time? Let's say every 5 days, or every Sunday at 4AM.

Thank you!

**Comment by gAlleb** [comment]
Hi @unudelaradio. Could I ask you why do you need this?

**Reply by unudelaradio** [reply]
I'm on Azuracast v20.2 and experiencing the old issue with glitches at the end of tracks, before mixing. The only way i know to solve this is restarting the AutoDJ. Then after one week the issue reappear. So i was thinking to plan an Auto Restart every 5 or 7 days, late at night. To avoid these glitches.

**Comment by LordHelmchen666** [comment]
You can do it with a cron job calling the API like this
`15 1,13 * * *    root    curl -X 'POST' 'https://YOURURL/api/station/YOURSTATION/backend/restart' -H 'accept: application/json' -H 'X-API-Key: YOUR_API_KEY' -d ''`
This would be everday at 01.15 and 13.15

**Reply by unudelaradio** [reply]
Took me some time to learn how to create a cron job in Linux, but now i learned, and the method you suggested works fine for me. Lots of thanks!

**Reply by gAlleb** [reply]
Still looks like a bad workaround. LS is supposed to run smoothly for a long time.

>  experiencing the old issue with glitches at the end of tracks, before mixing.

Is it a known issue? What glitches are we talking about?

**Reply by unudelaradio** [reply]
We've also talked about this issue here: https://github.com/AzuraCast/AzuraCast/discussions/6894#discussion-6136816

To be clear, i'm not confusing this behavior with the one caused by High CPU, which causes glitches on all stations at the same time and can't be solved with a simple restart of the station.

I also know about past issues with pops and glitches at the start of songs, which are now solved.

But this issue seems to persist. After a while, maybe 5 or 7 days, on some stations at the end of almost every song, a small pause appears, right before the jingle starts, right before the mixing process is about to start.

**Reply by gAlleb** [reply]
I see. 

I think you should wait a little for `LS 2.3.0 final release` and Azuracast update with this release. A `lot` of internal stuff has been changed and I think this issue should be resolved there too. If not - then we need to revitalize that old issue.

Or you can already try and test 2.3.0 within Azuracast. Somewhere on a **test enviroment** (or with a full backup):

```ruby
./docker.sh update-self
yes | ./docker.sh setup-release dev-ls2.3.x
./docker.sh update
```

I believe this issue should be resolved there.
